src/toil/jobStores/utils.py

```diff
@@ -220,17 +217,11 @@ def __enter__(self):
         return self.readable
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        try:
-            if exc_type is None:
-                if self.thread is not None:
-                    # reraises any exception that was raised in the thread
-                    self.thread.join()
-        finally:
-            self.readable.close()
-            # The responsibility for closing the writable end is generally that of the writer
-            # thread. To cover the small window before the writer takes over we also close it here.
-            writable_fh = self.writable_fh
-            if writable_fh is not None:
-                # FIXME: This is still racy. The writer thread could close it now, and someone
-                # else may immediately open a new file, reusing the file handle.
-                os.close(writable_fh)
+        # Close the read end of the pipe. The writing thread may
+        # still be writing to the other end, but this will wake it up
+        # if that's the case.
+        self.readable.close()
+        if exc_type is None:
```

Comment:
This is from the original code--I didn't want to screw with the behavior too much. That said, I also thought this was a bit weird.

I can only guess, but I think the logic here is that joining the thread may raise a new exception, masking the one from the main thread. The other thread is, in theory, doomed to complete or die at this point, so we shouldn't end up leaking the thread.

How about this? We could just get rid of this if, and wrap the join in a try/except that reraises the child exception if exc_type is None and `pass`es otherwise. I think that would be cleaner, and it would guarantee that we don't leak threads.